File: ExtranetApps.Api/Models/ClientesGestion.cs
using System;

namespace ExtranetApps.Api.Models
{
    public class ClientesGestion
    {
        public int Id { get; set; }
        public int ClienteId { get; set; }

        public TipoGestion TipoGestion { get; set; }
        public string Observaciones { get; set; }
        //greencore.

        public ClienteAdjunto adjunto { get; set; }
        public DateTime Fecha { get; set; }
        public DateTime? FechaRecontacto { get; set; }


        public virtual string FullDescription
        {
            get
            {
                return TipoGestion != null ? string.Format("{0} - {1}", TipoGestion.Descripcion, Observaciones) : Observaciones;
            }
        }
    }

    public class ClienteAdjunto
    {

        public int Id { get; set; }
        public int ClienteId { get; set; }
        public int ClienteGestionId { get; set; }
        //private Panel.AdjuntosClasificaciones clAdjuntoClasificacionId;
        //private Panel.TiposAdjuntos clTipoAdjuntoId;
        //public DateTime FecVencimiento { get; set; }
        //public string Name { get; set; }
        public string Archivo { get; set; }
    }
}

File: ExtranetApps.Site/src/app/models/clientes-gestion.ts
import { listable } from "./listable.model";
import { ClienteAdjuntos } from "./clienteAdjuntos.model";

export class ClientesGestion {
    Id: number;
    clienteId: number;
    tipoGestion: listable;
    observaciones: string;
    // pdfgestion:byte[];
    fecha: Date | string;
    fechaRecontacto: Date | string;
    fulldescription: string;
    adjunto: ClienteAdjuntos;

    constructor
        (
            Id: number = 0,
            clienteId: number = 0,
            tipoGestion: listable = new listable("", ""),
            observaciones: string = "",
            fecha: Date | string = new Date(),
            fechaRecontacto: Date | string = new Date(),
            fulldescription: string = "",
            adjunto: ClienteAdjuntos = new ClienteAdjuntos()
        ) {
        this.Id = Id;
        this.clienteId = clienteId;
        this.tipoGestion = tipoGestion;
        this.observaciones = observaciones;
        this.fecha = fecha;
        this.fechaRecontacto = fechaRecontacto;
        this.fulldescription = fulldescription;
        this.adjunto = adjunto;
    }
}